### PR TITLE
add unit test suite, include unit test for broken is_dir method

### PR DESCRIPTION
**Pull request by boegel**
depends on https://github.com/hpcugent/vsc-base/pull/140

only includes a single unit test for now, but this is already a nice example of why it's interesting to have them:

```
PYTHONPATH=$HOME/work/vsc-base/lib:$PWD/lib:$PYTHONPATH python test/posix.py
F
======================================================================
FAIL: test_is_dir (__main__.PosixTest)
Tests for is_dir method.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/posix.py", line 37, in test_is_dir
    self.assertEqual(self.po.is_dir(os.environ['HOME']), True)
AssertionError: None != True

----------------------------------------------------------------------
Ran 1 test in 0.002s

FAILED (failures=1)
```

### CODE REVIEW

**Comment by boegel** [comment]
ping on this, important enough to revive?

**Comment by stdweird** [comment]
get rid of the vim thingie, and i still ahven't thought about the `is_dir` issue :smile:

File: lib/vsc/filesystem/posix.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def make_symlink(self, target, obj=None, force=True):
     def is_dir(self, obj=None):
         """Check if it is a directory"""
         obj = self._sanity_check(obj)
-        # do symlinks count ?
+        return os.path.isdir(obj)
```
Comment (stdweird) [inline lib/vsc/filesystem/posix.py]: so add in the docstring that a symlink to a directory is considered a directory (we don't use the `is_dir` anywhere, so this change is harmless)

**Comment by boegel** [comment]
revived (synced with master), remarks fixed

**Pull request merged**